### PR TITLE
Build bauba_oep framework

### PR DESCRIPTION
**Pull request by sdk-banuba**

### CODE REVIEW

File: toolchain/pack_framework.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+mkdir -p "$frameworkOutputPath/Versions/A/Headers"
+
+# # Link the "Current" version to "A"
+/bin/ln -sfh A "$frameworkOutputPath/Versions/Current"
+/bin/ln -sfh Versions/Current/Headers "$frameworkOutputPath/Headers"
+/bin/ln -sfh "Versions/Current/lib$1.a" "$frameworkOutputPath/$1"
+
+# #copy 
+/bin/mv "$libOutputPath/lib$1.a" "$frameworkOutputPath/Versions/A/" #${libraryName}
+# /bin/mv "${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.framework/Versions/A/${PRODUCT_NAME}.a" "${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.framework/Versions/A/${PRODUCT_NAME}"
```
Comment (dmitrystoyanovsky) [inline toolchain/pack_framework.sh:21]: remove

File: toolchain/pack_framework.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+rm -rf "$frameworkOutputPath"
+mkdir -p "$frameworkOutputPath/Versions/A/Headers"
+
+# # Link the "Current" version to "A"
+/bin/ln -sfh A "$frameworkOutputPath/Versions/Current"
+/bin/ln -sfh Versions/Current/Headers "$frameworkOutputPath/Headers"
+/bin/ln -sfh "Versions/Current/lib$1.a" "$frameworkOutputPath/$1"
+
+# #copy 
+/bin/mv "$libOutputPath/lib$1.a" "$frameworkOutputPath/Versions/A/" #${libraryName}
```
Comment (dmitrystoyanovsky) [inline toolchain/pack_framework.sh:20]: please remove #${libraryName}

**Review by dmitrystoyanovsky — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 3]
```diff
@@ -91,16 +107,21 @@ set_target_properties(${APP_NAME} PROPERTIES
...
 
-copy_third(${APP_NAME})
+set_target_properties(${APP_NAME} PROPERTIES XCODE_ATTRIBUTE_FRAMEWORK_SEARCH_PATHS "${SDK_FRAMEWORK_PATH}/$<IF:$<CONFIG:Debug>,debug,release>")
+set_target_properties(${APP_NAME} PROPERTIES XCODE_ATTRIBUTE_HEADER_SEARCH_PATHS "$<TARGET_FILE_DIR:${BANUBA_OEP_FRAMEWORK}>/${BANUBA_OEP_FRAMEWORK}.framework/Headers")
+set_target_properties(${APP_NAME} PROPERTIES XCODE_ATTRIBUTE_OTHER_LDFLAGS "-F$<TARGET_FILE_DIR:${BANUBA_OEP_FRAMEWORK}> -lc++")
+set_target_properties(${APP_NAME} PROPERTIES XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_ARC "YES")
+
+target_link_libraries(${APP_NAME} ${OTHER_LDFLAGS})
+
+copy_third(${APP_NAME})
```
[the remark this answers is not in the document]
Comment (sdk-banuba) [reply CMakeLists.txt:127]: Yes that's right

File: CMakeLists.txt [hunk 4]
```diff
@@ -13,27 +13,30 @@ set(BNB_THIRD_FOLDER ${CMAKE_CURRENT_LIST_DIR}/third)
...
+
+# some magic to add these files to copy phase
+set(SDK_FRAMEWORK_PATH "${CMAKE_CURRENT_LIST_DIR}/bnb_sdk/mac")
+set(FullEPFrameworkPath "${SDK_FRAMEWORK_PATH}/$(CONFIGURATION)/BanubaEffectPlayer.framework")
+add_custom_command(OUTPUT "${FullEPFrameworkPath}" COMMAND "")
+set(FullPPFrameworkPath "${SDK_FRAMEWORK_PATH}/$(CONFIGURATION)/BanubaPostprocess.framework")
+add_custom_command(OUTPUT "${FullPPFrameworkPath}" COMMAND "")
+
+
+#set_source_files_properties(${FullEPFrameworkPath} PROPERTIES MACOSX_PACKAGE_LOCATION Resources)
```
Comment (dmitrystoyanovsky) [inline CMakeLists.txt:36]: remove also

File: README.md [hunk 5]
```diff
@@ -5,22 +5,44 @@ Quick start examples for integrating [Banuba SDK on macos](https://docs.banuba.c
 1. Get the latest Banuba SDK archive for MacOS/Windows and the client token. Please fill out our form at [form at banuba.com](https://www.banuba.com/face-filters-sdk) website, or contact us via [info@banuba.com](mailto:info@banuba.com).
 2. Copy `bnb_viewer_standalone/bnb_sdk/` into the `OEP_macos/bnb_sdk` dir:
     `bnb_viewer_standalone/bnb_sdk/` => `OEP_macos/bnb_sdk`
-3. Copy `bnb_viewer_standalone/resources/` files into the `OEP_macos/resources` dir:
-    `bnb_viewer_standalone/resources/` => `OEP_macos/resources`
-4. Copy `bnb_viewer_standalone/third` files into the `OEP_macos/third` dir:
-    `bnb_viewer_standalone/third/` => `OEP_macos/third`
-5. Copy and Paste your client token into the appropriate section of `OEP_macos/ViewController.swift`
-6. Generate project files by executing the following commands:
+3. Copy `bnb_viewer_standalone/third` files into the `OEP_macos/third` dir:
```
Comment (dmitrystoyanovsky) [inline README.md:8]: now contradictionary - copy third but below glad.

Comment (dmitrystoyanovsky) [reply README.md:8]: Easier to leave as it was before.

File: README.md [hunk 6]
```diff
@@ -5,22 +5,44 @@ Quick start examples for integrating [Banuba SDK on macos](https://docs.banuba.c
...
+4. Copy the `bnb_sdk` to your application direcory.
+5. Copy the `effects` from `resources` to your application direcory.
+6. Add references to frameworks in your project settings in the `General` tab in `Frameworks, Libraries, and Embedded Content`:
+    - Accelerate
+    - OpenGL
+    - BanubaEffectPlayer
+    - BanubaPostprocess
+    - banuba_oep
+![Alt text](/resources/images/2DB863E6-8769-43CF-BAD9-21872C4147DA_4_5005_c.jpeg?raw=true "Title")
+7. Add paths to the headers `banuba_oep.framework/Headers` in the `Build Settings` tab `Header Search...` option
```
Comment (dmitrystoyanovsky) [inline README.md:37]: Header Search... means that you should use complete name I don't remember it

File: README.md [hunk 7]
```diff
@@ -5,22 +5,44 @@ Quick start examples for integrating [Banuba SDK on macos](https://docs.banuba.c
...
+![Alt text](/resources/images/EE331F32-85E8-4FDC-8818-3640F0315FEB_4_5005_c.jpeg?raw=true "Title")
+8. Add linker option `-lc++` to in the `Build Settings` tab
```
Comment (dmitrystoyanovsky) [inline README.md:39]: Which option name to update

File: ViewController.swift [hunk 8]
```diff
@@ -26,12 +26,12 @@ class ViewController: NSViewController, AVCaptureVideoDataOutputSampleBufferDele
     private let renderWidth: UInt = 1280
     private let renderHeight: UInt = 720
     private var effectLoaded = false
-    private let token = <<#place your token here#>>
+    private let token = "WMHjeKZfQ8WJ0Z/zN2TAiTRKfEC8pkAjGbiOTzWJ+EKEs13Mtx58NWTUF6yDH+lup69EPkz0PqE="
```
Comment (dmitrystoyanovsky) [inline ViewController.swift:29]: remove token

**Review by dmitrystoyanovsky — APPROVED** [review]

**Pull request merged**